doc/en/how-to/capture-warnings.rst

```diff
@@ -300,6 +300,32 @@ filter at the end of the test, so no global state is leaked.
 
 .. _recwarn:
```

Comment:
This is a [cross-reference label](https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#cross-referencing-arbitrary-locations) referring to the `Recording warnings` section, so it should be kept close to it.

We should also create a new cross reference label here that can be used instead of the full URL in the changelog:

```suggestion
.. _warns-use-cases:
```